### DISCUSSION TITLE
Source file with blank in path

### DISCUSSION THREAD

**Discussion by fellnerse** [nushell/nushell#5044] [Q&A] [answered]
Hello!

I would like to source a file in my `config.nu` (just as suggestion, it also would be awesome if it would be possible to source directly from the scripts folder as well here) with this code:

```
let file = (echo $'($env.NU_LIB_DIRS | path join x86.nu)' )
echo $file
cat $file
#source $file
```

The output in a new shell is:

```
/Users/xxx/Library/Application Support/nushell/scripts/x86.nu
# open emulated nushell (this means all commands run with this shell are emulated for x86)
export def x86 [
            ...argv  # additional cli arguments
        ] {
    arch -x86_64 sh -c 'nu' $argv;
}
```

But when I uncomment the source line from above it says it can not find the the file. I suspect this is due to the blank in the path? Do you know how I can do this? Thanks!

Edit:
I understand that this issue is related to: https://github.com/nushell/nushell/discussions/4905
But do you offer a preferred way of sourcing files in the `config.nu`? listing full paths feels a bit weird and I also do not want to source this all the time from hand in my terminal

**Comment by kubouch** [comment, marked as answer]
Yeah, we do not support sourcing from dynamic file names, see https://www.nushell.sh/book/thinking_in_nushell.html#parsing-and-evaluation-are-different-stages.

We plan to hardcode the default NU_LIB_DIRS pointing at `<nushell-config-dir>/scripts` so that it would be available within `config.nu`. Right now, you need to use absolute path.

**Reply by fellnerse** [reply]
Thanks for clarification! Can i help somehow on this? Or what is the status?
do you think it also would make sense to have a default folder containing files that will automatically be sourced?
So one that is available for sourcing and one that is sourced during `config.nu`

**Reply by kubouch** [reply]
The status is I need to find time for it :D. I wanted to connect this with an improvement on how we collect environment variables on Nushell startup. It would involve changing [this function](https://github.com/nushell/nushell/blob/d89ad4fafd1c7bc458931d51ba7fa4aac77a9d0c/crates/nu-cli/src/util.rs#L100) to save into our nuon format, rather than the custom "name=value". Once we have that, we can add NU_LIB_DIRS into the collected nuon easily. I'm not sure how familiar you are with Nushell or Rust, it might be a bit involved since it requires knowing a bit about the parser. But we certainly welcome help all around the project. If you're new to Nushell and want to contribute, I would recommend starting with understanding how the built-in commands work and from there you can reach deeper into the source code.

I would personally be against automatically sourcing files from a folder since you can do that in `config.nu` and once the hard-coded NU_LIB_DIRS is in, it should be more user friendly.
